gtsam/base/SparseMatrix.h

```diff
@@ -0,0 +1,68 @@
+/* ----------------------------------------------------------------------------
+
+ * GTSAM Copyright 2010, Georgia Tech Research Corporation,
+ * Atlanta, Georgia 30332-0415
+ * All Rights Reserved
+ * Authors: Frank Dellaert, et al. (see THANKS for the full author list)
+
+ * See LICENSE for the license information
+
+ * -------------------------------------------------------------------------- */
+
+/**
+ * @file    SparseMatrix.h
+ * @brief   Interface utilities with Eigen's SparseMatrix representation
+ * @author  Gerry Chen
+ */
+
+#pragma once
+
+#include "gtsam/linear/GaussianFactorGraph.h"
```

Comment:
base cannot know about linear :-/ the directories in GTSAm form a directed graph.
I think all these are utilities for GaussianFactorGraph functions, right? Then why not make them static functions *inside* GaussianFactorGraph.cpp? Only expose the type (which you exposed in Matrix.h) in the GaussianFactorGraph.h header.